pkg/doc/bbs/bbs12381g2pub/internal/hash_to_field.go

```diff
@@ -0,0 +1,83 @@
+/*
+Taken from https://github.com/kilic/bls12-381
+Changes:
+1) pass hash function as input for hashToFpXMD() - e.g. don't stick on SHA-256 only.
```

Comment:
Please structure as new code going to a new file, to minimize how much we change original sources.

Reply:
File renamed to `hash_to_field_custom.go` as all the functions inside it are rewritten by us.